nengo/tests/test_solvers.py

```diff
@@ -576,7 +576,41 @@ def test_nosolver_validation():
     NoSolver(values=[[1], [1]])
 
 
-def test_non_compositional_solver(Simulator):
+@pytest.mark.parametrize('solver', [
+    LstsqDrop(weights=True),
+    # LstsqL1(weights=True),
+    # NnlsL2(weights=True),
```

Comment:
I'm guessing these are meant to be un-commented?

Reply:
I uncommented them and the `NnlsL2` test fails... should we be concerned about this?

Reply:
Also, these tests are pretty slow on my machine, is it possible to lower the simulation run time to something more like 0.1?

Reply:
No, we should probably get rid of them. `LstsqL1` works, it's just slow. I couldn't figure a way to mark it slow without marking the whole test as slow. `NnlsL2` "works", but because it forces non-negative weights the communication channel doesn't work as well, so it fails the test the way it's written now.

Reply:
So do we have any tests that LstsqL1 is working correctly?  I notice that you changed it to non-compositional, which I assume means that it was incorrect previously, but no tests were failing.

Reply:
I think we have tests for all the solvers, but most of them are unit tests, not integration tests. So they wouldn't be affected by `compositional` being wrong.

It would be nice to have integration tests, but it's hard to have ones that really work. For example, with LstsqL1, you'd have to check that the sparsity of the final weights is as desired, to test whether `compositional` is set correctly. And it's not clear what the output sparsity should be given a particular regularization value. With LstsqDrop and Nnls, it's a little easier, since you can check for the desired sparsity or that the weights are non-negative, respectively, though as I said for Nnls, the weights don't work as well because they're non-negative.

One option would be to have a common test helper function that creates this network, and then have separate tests for each of the three compositional solvers that can check both the output signal and the weights specifically for that solver. Then we can mark LstsqL1 as slow without affecting the others. Let me know if you want me to do this.